### DISCUSSION TITLE
percussive wav file to test overall latency

### DISCUSSION THREAD

**Discussion by FelixTheKitten** [GrandOrgue/grandorgue#2018] [Help]
I would like to measure the total time between the directly measured key switch voltage step and a percussive sound produced by grand orgue.  I need help creating a "latency test" organ with just one stop consisting of a percussive wav file, for example a step function with rapid exponential decay ( same for any note ).  The step should be located as early as possible in the wav file.  If I mix the key switch voltage with the grand orgue output, the time between the two steps should give the total system latency.  I want to use this to accurately compare OS Ventura and OS Sonoma, because I suspect Sonoma has slower midi input.

**Comment by midimusic** [comment]
You could record a fairly high (e.g. top C = midi 72) piano note and use that as the sample. No looping is needed. 

You do not necessarily have to make a special organ. Just substitute your piano sample for one of the organ's original samples.

I have previously made latency measurements of various versions GO and I think you will find the total latency lies between 70 and 100ms.

csw900
